api/org/commcare/api/persistence/SqlIndexedStorageUtility.java

```diff
@@ -0,0 +1,418 @@
+            preparedStatement = SqlHelper.prepareTableSelectStatement(c, this.tableName,
+                    new String[]{fieldName}, new String[]{(String)value}, prototype.newInstance());
+            if (preparedStatement == null) {
+                return null;
+            }
+            ResultSet rs = preparedStatement.executeQuery();
+                    Vector<Integer> ids = new Vector<>();
+            while (rs.next()) {
+                ids.add(rs.getInt(org.commcare.modern.database.DatabaseHelper.ID_COL));
+            }
+            return ids;
+        } catch (InstantiationException | IllegalAccessException | SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                if (c != null) {
+                    c.close();
+                }
+                if (preparedStatement != null) {
+                    preparedStatement.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+
+        }
+
+        return null;
+    }
+
+    @Override
+    public T getRecordForValue(String fieldName, Object value)
+            throws NoSuchElementException, InvalidIndexException {
+        Connection c = null;
+        PreparedStatement preparedStatement = null;
+        try {
+            c = this.getConnection();
+            preparedStatement =
+                    SqlHelper.prepareTableSelectStatement(c, this.tableName,
+                            new String[]{fieldName}, new String[]{(String)value},
+                            prototype.newInstance());
+            if (preparedStatement == null) {
+                throw new NoSuchElementException();
+            }
+            ResultSet rs = preparedStatement.executeQuery();
+            if (!rs.next()) {
+                throw new NoSuchElementException();
+            }
+            byte[] mBytes = rs.getBytes(org.commcare.modern.database.DatabaseHelper.DATA_COL);
+            return readFromBytes(mBytes);
+        } catch (SQLException | InstantiationException |
+                IllegalAccessException | NullPointerException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                if (c != null) {
+                    c.close();
+                }
+                if (preparedStatement != null) {
+                    preparedStatement.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return null;
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#add(org.javarosa.core.util.externalizable.Externalizable)
+     */
+    public int add(T e) throws StorageFullException {
+        this.write(e);
+        return 1;
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#close()
+     */
+    public void close() {
+        // TODO Auto-generated method stub
```

Comment:
Could you replace this auto-generated comment with one explaining that this method is only relevant for J2ME code? At least I think that is the only time it has a non-empty implementation